File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        if (points.length == 1) {
            return 1;
        }
        Arrays.sort(points, (a, b) -> Integer.compare(a[0], b[0]));
        int ans = 1;
        int end = points[0][1];
        for(int[] point : points) {
            if(point[0] <= end) {
                end = Math.min(end, point[1]);
            } else {
                end = point[1];
                ans++;
            }
        }
        return ans;
    }
}

File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key = lambda x: x[1])

        arrows = 0
        i = 0
        while i < len(points):
            point = points[i]
            end = point[1]
            while i + 1 < len(points) and points[i + 1][0] <= end:
                i += 1
            arrows += 1
            i += 1
        return arrows
         